packages/eas-cli/src/commands/upload.ts

```diff
@@ -107,10 +122,24 @@ export default class BuildUpload extends EasCommand {
       { distribution: DistributionType.Internal, fingerprintHash: fingerprint, developmentClient }
     );
 
-    Log.withTick(`Here is a sharable link of your build: ${getBuildLogsUrl(build)}`);
+    if (jsonFlag) {
+      printJsonOnlyOutput({ url: getBuildLogsUrl(build) });
+    } else {
+      Log.withTick(`Shareable link to the build: ${getBuildLogsUrl(build)}`);
+    }
```

Comment:
Preference to organize code like this, because if you add on extra logic later (e.g., printing other logs), having the early return keeps the structure clean and prevents deeply nested conditionals. It also separates the JSON-only output path from the interactive logging path.

```suggestion
    if (jsonFlag) {
      printJsonOnlyOutput({ url: getBuildLogsUrl(build) });
      return;
    } 
    
    Log.withTick(`Shareable link to the build: ${getBuildLogsUrl(build)}`);
    
```